### ISSUE TITLE
H5File.Open() on an already opened file fails on .net framework

### ISSUE THREAD

**Issue by audunska** [Apollo3zehn/PureHDF#123]
On .net framework, the following code fails with `System.ArgumentException: 'The path has an invalid format'` (PureHdf version 1.0.0):
```c#
var file = System.IO.File.OpenRead(path);
H5File.Open(file);
```
The reason seems to be [this line](https://github.com/Apollo3zehn/PureHDF/blob/ddb83dc9956697829dd5f92ab2dcad19e02ed7e5/src/PureHDF/API.Reading/H5File.cs#L88), which sets the file path to an empty string.

When we later [call GetDirectoryName on that path](https://github.com/Apollo3zehn/PureHDF/blob/ddb83dc9956697829dd5f92ab2dcad19e02ed7e5/src/PureHDF/VOL/Native/API.Reading/NativeFile.cs#L28), this throws an exception on the empty path.

Indeed, [GetDirectoryName throws on empty paths](https://learn.microsoft.com/en-us/dotnet/api/system.io.path.getdirectoryname?view=netframework-4.8.1) in .net framework.

I realize that version 2 has dropped support for .net framework. However, my codebase can't be upgraded quite yet. The fix ought to be simple - just avoid calling `GetDirectoryName` on empty paths. Would you accept a pull request to fix this in v1.0?

**Comment by Apollo3zehn** [comment]
Thanks for bringing this up. Please go ahead with the pull request :-) I am in holidays so I can merge it earliest next week.

Why did you change the issue to title to "already opened file"? What does this mean?

**Comment by audunska** [comment]
I just mean that we pass a file to `H5File.Open`, and not a path.

**Comment by audunska** [comment]
Since this problem only exists pre .net 6.0, a fix is not necessary on the master branch, but only in v.1.0.0. So how should I go about making the pull request? Do you want to make a branch for v1.0.0 fixes, or would you like the change to apply to the master branch too?

FWIW, the fix is just a one-line diff:
```diff
diff --git a/src/PureHDF/VOL/Native/API.Reading/NativeFile.cs b/src/PureHDF/VOL/Native/API.Reading/NativeFile.cs
index 0b38ab8..080463a 100644
--- a/src/PureHDF/VOL/Native/API.Reading/NativeFile.cs
+++ b/src/PureHDF/VOL/Native/API.Reading/NativeFile.cs
@@ -25,7 +25,7 @@ public class NativeFile : NativeGroup, IDisposable
         bool deleteOnClose) : base(context, reference, header)
     {
         Path = absoluteFilePath;
-        FolderPath = System.IO.Path.GetDirectoryName(absoluteFilePath);
+        FolderPath = absoluteFilePath.Length > 0 ? System.IO.Path.GetDirectoryName(absoluteFilePath) : string.Empty;
         _deleteOnClose = deleteOnClose;
     }
```

**Comment by audunska** [comment]
But then I need a branch to PR against. There is no branch for the 1.0 line, or am I missing something?

**Comment by Apollo3zehn** [comment]
Oh you are right, there is only a v1.0.0 tag. Then please create ist against master and I will change it later. Sorry for the confusion

**Comment by Apollo3zehn** [comment]
There is now the `v1` branch.

**Comment by audunska** [comment]
I'll make a pr tomorrow when at work then.

**Comment by Apollo3zehn** [comment]
Closed via #126 

Thanks!
